### PR TITLE
virsh_domiftune: Support 'config' option with mac address

### PR DESCRIPTION
**Pull request by hahakiki2010**
1) Add 'config' option test
2) Add set_domiftune by interface name or by MAC address
3) Add outside boundary test

Signed-off-by: Yan Li <yannli@redhat.com>

### CODE REVIEW

**Comment by hahakiki2010** [comment]
Test passed
```
(virtenv_py3) [root@localhost src]# avocado run --vt-type libvirt virsh.domiftune..outside_boundary
JOB ID     : d07fe3789efeea68db0429b26780bf96ac5e33ee
JOB LOG    : /root/avocado/job-results/job-2018-07-30T16.18-d07fe37/job.log
 (1/8) type_specific.io-github-autotest-libvirt.virsh.domiftune.negative_testing.set_domif_parameter.running_guest.change_inbound.options.live.outside_boundary: PASS (18.72 s)
 (2/8) type_specific.io-github-autotest-libvirt.virsh.domiftune.negative_testing.set_domif_parameter.running_guest.change_inbound.options.config.outside_boundary: PASS (25.14 s)
 (3/8) type_specific.io-github-autotest-libvirt.virsh.domiftune.negative_testing.set_domif_parameter.running_guest.change_inbound.options.current.outside_boundary: PASS (34.40 s)
 (4/8) type_specific.io-github-autotest-libvirt.virsh.domiftune.negative_testing.set_domif_parameter.running_guest.change_inbound.options.none.outside_boundary: PASS (24.11 s)
 (5/8) type_specific.io-github-autotest-libvirt.virsh.domiftune.negative_testing.set_domif_parameter.running_guest.change_outbound.options.live.outside_boundary: PASS (24.22 s)
 (6/8) type_specific.io-github-autotest-libvirt.virsh.domiftune.negative_testing.set_domif_parameter.running_guest.change_outbound.options.config.outside_boundary: PASS (24.35 s)
 (7/8) type_specific.io-github-autotest-libvirt.virsh.domiftune.negative_testing.set_domif_parameter.running_guest.change_outbound.options.current.outside_boundary: PASS (24.16 s)
 (8/8) type_specific.io-github-autotest-libvirt.virsh.domiftune.negative_testing.set_domif_parameter.running_guest.change_outbound.options.none.outside_boundary: PASS (34.61 s)
RESULTS    : PASS 8 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 272.39 s

(virtenv_py3) [root@localhost src]# avocado run --vt-type libvirt virsh.domiftune.positive_testing.set_domif_parameter.running_guest.mac_address.config
JOB ID     : 2a29322ddc675add7343364a2768571dcd5b3c58
JOB LOG    : /root/avocado/job-results/job-2018-07-30T16.32-2a29322/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.domiftune.positive_testing.set_domif_parameter.running_guest.mac_address.config: PASS (52.21 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 62.46 s
```

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py [hunk 1]
```diff
@@ -308,18 +314,34 @@ def run(test, params, env):
...
     interface = []
+
     if vm and not vm.is_alive():
         vm.start()
+
     if vm and vm.is_alive():
         virt_xml_obj = vm_xml.VMXML(virsh_instance=virsh)
         interface = virt_xml_obj.get_iface_dev(vm_name)
-        logging.debug("the interface is %s" % interface[0])
+        if_mac = interface[0]
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py]: I don't you need so many changes, the logic is very simple:
```
if interface_ref:
interface.append(if_mac)
```
other changes are not necessary

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_domiftune.cfg [hunk 2]
```diff
@@ -143,6 +156,7 @@
                                                     options =
                         - shutoff_guest:
                             start_vm = "no"
+                                interface_ref = "mac"
```
Comment (balamuruhans) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_domiftune.cfg]: indentation issue

File: libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py [hunk 3]
```diff
@@ -308,18 +314,34 @@ def run(test, params, env):
...
+        # Get interface name
+        vmxml = virt_xml_obj.new_from_dumpxml(vm_name)
+        if_node = vmxml.get_iface_all().get(if_mac)
+        if_name = if_node.find('target').get('dev')
+
+    if interface_ref == "name":
+        interface = if_name
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/domain/virsh_domiftune.py]: if no use later for if_name, I think the above " get interface name " part should be in this "if"

**Comment by balamuruhans** [comment]
@chunfuwen, The changes suggested by @kylazhang and indentation issue are not resolved yet.

**Pull request merged**